include/xtensor-signal/find_peaks.hpp

```diff
@@ -0,0 +1,725 @@
+            xt::xarray<double> _prominence_data;
+            //check if prominance data is being supplied
+
+            //Calculate prominence if not supplied and use wlen if supplied.
+            //check if wlen is an acceptable parameter
+            auto wlen_safe = detail::arg_wlen_as_expected(wlen);
+            auto [prominences, left_bases, right_bases] = detail::peak_prominences(x, peaks, wlen_safe);
+            return detail::peak_widths(x, peaks, rel_height, prominences, left_bases, right_bases);
+        }
+
+        /**
+         * @brief finds all peaks and applies filters to the peaks based on the parameters provided.
+         * @details portions of this function are unimplemented and will throw a compiler error if features requested are not available.
+         * @param 1D data vector.
+         * @param minimum height thresholds to be filtered. Only minimum heights are implemented.
+         * @param threshold unimplemented.
+         * @param distance unimplemented.
+         * @param prominence unimplemented.
+         * @param width defines the minimum width for eligable points. if a second parameter is applied a max width is also applied.
+         * @param wlen defined wlen for calculating prominence.
+         * @param rel_height used in defining the width of peaks based on the location. Defaults to .5  or 6dB.
+         * @param plateau_size unimplemented.
+         */
+        class find_peaks
+        {
+        public:
+            using height_t = 
+            std::variant<
+                float,
+                std::pair<float, float>, 
+                xt::xtensor<float, 1>,
+                std::pair<xt::xtensor<float,1>, xt::xtensor<float,1>>
+            >;
+
+            using threshold_t =
+            std::variant<
+                float,
+                std::pair<float, float>, 
+                xt::xtensor<float, 1>,
+                std::pair<xt::xtensor<float,1>, xt::xtensor<float,1>>
+            >;
+
+            using prominence_t =
+            std::variant<
+                float,
+                std::pair<float, float>, 
+                xt::xtensor<float, 1>,
+                std::pair<xt::xtensor<float,1>, xt::xtensor<float,1>>
+            >;
+
+            using width_t =
+            std::variant<
+                float,
+                std::pair<float, float>, 
+                xt::xtensor<float, 1>,
+                std::pair<xt::xtensor<float,1>, xt::xtensor<float,1>>
+            >;
+
+            using plateau_t =
+            std::variant<
+                size_t,
+                std::pair<size_t, size_t>, 
+                xt::xtensor<size_t, 1>,
+                std::pair<xt::xtensor<size_t,1>, xt::xtensor<size_t,1>>
+            >;
+
+            find_peaks() : _rel_height(.5)
+            {
+    
+            }
+            find_peaks& set_height(height_t height)
+            {
+                _height = std::make_optional(height);
```

Comment:
Parameter height can be safely moved here. Same remark applies to the setters below.